### PR TITLE
MAINT: optimize.newton: avoid error with complex `x0`

### LINKED ISSUE #10103: BUG: optimize: newton raises exception with complex x0
Here's an example where `newton` crashes unexpectedly (using scipy version `'1.4.0.dev0+b4bfa9d'`):

```
In [3]: from scipy.optimize import newton                                                                                                                            

In [4]: def f(z): 
   ...:     return z - 1 
   ...:                                                                                                                                                              

In [5]: newton(f, 1+1j)                                                                                                                                              
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-5306645b6b37> in <module>
----> 1 newton(f, 1+1j)

~/miniconda36/lib/python3.6/site-packages/scipy-1.4.0.dev0+b4bfa9d-py3.6-macosx-10.7-x86_64.egg/scipy/optimize/zeros.py in newton(func, x0, fprime, args, tol, maxiter, fprime2, x1, rtol, full_output, disp)
    323             eps = 1e-4
    324             p1 = x0 * (1 + eps)
--> 325             p1 += (eps if p1 >= 0 else -eps)
    326         q0 = func(p0, *args)
    327         funcalls += 1

TypeError: '>=' not supported between instances of 'complex' and 'int'
```

### PR DESCRIPTION
**Pull request by mdhaber**
#### Reference issue
Closes gh-10103

#### What does this implement/fix?
gh-10103 reported an error when a Python complex `x0` was passed to `newton`. A comment revealed that the problem disappears when `x0` is a NumPy complex value, so this PR converts `x0` to a NumPy complex dtype to fix the problem.

### CODE REVIEW

File: scipy/optimize/tests/test_zeros.py [hunk 1]
```diff
@@ -836,3 +836,15 @@ def f(x):
     res = res if rs_interface else res[1]
     assert res.converged
     assert_allclose(res.root, 0, atol=1e-8)
+
+
+def test_newton_complex_gh10103():
+    # gh-10103 report a problem with `newton` and complex x0. Check that this
+    # is resolved.
```
Comment (mdhaber) [inline scipy/optimize/tests/test_zeros.py:843]:
```suggestion
    # gh-10103 reported a problem with `newton` and complex x0.
    # Check that this is resolved.
```

**Comment by mdhaber** [comment]
refguide_asv_check failure was in main; fixed by gh-17663.

File: scipy/optimize/tests/test_zeros.py [hunk 2]
```diff
@@ -836,3 +836,15 @@ def f(x):
     res = res if rs_interface else res[1]
     assert res.converged
     assert_allclose(res.root, 0, atol=1e-8)
+
+
+def test_newton_complex_gh10103():
+    # gh-10103 reported a problem with `newton` and complex x0.
```
Comment (pvanmulbregt) [inline scipy/optimize/tests/test_zeros.py:842]: Perhaps expand the description to something like:
```
# gh-10103 reported a problem with `newton` and complex x0, no fprime and no x1.
```

**Review by pvanmulbregt — CHANGES_REQUESTED** [review]
LGTM apart from the description of the new test.

File: scipy/optimize/tests/test_zeros.py (hunk 2 above)
Comment (mdhaber) [reply scipy/optimize/tests/test_zeros.py]: Thanks. Done!

**Review by pvanmulbregt — APPROVED** [review]

**Review by tupui — APPROVED** [review]
LGTM and it's testing both code paths 👍 Thanks Matt and Paul for the review.

**Pull request merged**